scrapy/http/request/__init__.py

```diff
@@ -57,8 +58,10 @@ def _set_url(self, url):
 
         s = safe_url_string(url, self.encoding)
         self._url = escape_ajax(s)
-
-        if ':' not in self._url:
+        # add better url scheme verification @stephenhay
+        pattern = "https://[^\s/$.?#].[^\s]*|^http://[^\s/$.?#].[^\s]*"
```

Comment:
I think ``http`` and ``https`` are not the only possible schemes, in fact any scheme is possible with a custom downloader that supports it. I think this check is simple on purpose, it's main goal is to provide an error message for a common error, so it's fine if it's not exact - stuff will error out later in any case.
@soubenz are you trying to solve a specific use-case when an existing check was not enough?

Reply:
I think Scrapy should at least show a warning log message when the url doesn't match the most common schemes for webpages http & https. That is of course in case of using the default scrapy downloader.